File: src/hasoftware-library/src/hasoftware/api/messages/LoginRequest.java
package hasoftware.api.messages;

import hasoftware.api.FunctionCode;
import hasoftware.api.Message;
import hasoftware.cdef.CDEFMessage;

public class LoginRequest extends Message {

    private String _username;
    private String _password;

    public LoginRequest() {
        super(FunctionCode.Login, 0);
    }

    public LoginRequest(CDEFMessage cdefMessage) {
        super(cdefMessage);
        _username = cdefMessage.getAsciiL();
        _password = cdefMessage.getAsciiL();
    }

    @Override
    public void encode(CDEFMessage cdefMessage) {
        super.encode(cdefMessage);
        cdefMessage.putAsciiL(_username);
        cdefMessage.putAsciiL(_password);
    }

    public String getUsername() {
        return _username;
    }

    public void setUsername(String username) {
        _username = username;
    }

    public String getPassword() {
        return _password;
    }

    public void setPassword(String password) {
        _password = password;
    }
}

File: src.net/hasoftware-library/Messages/LoginRequest.cs
using hasoftware.Api;
using hasoftware.Cdef;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace hasoftware.Messages
{
    public class LoginRequest : Message
    {
        public string Username {get;set;}
        public string Password {get;set;}

        public LoginRequest()
            :base(Api.FunctionCode.Login, 0)
        {
        }

        public LoginRequest(CdefMessage cdefMessage)
            :base(cdefMessage)
        {
            Username = cdefMessage.GetAsciiL();
            Password = cdefMessage.GetAsciiL();
        }

        public override void Encode(CdefMessage cdefMessage) {
            base.Encode(cdefMessage);
            cdefMessage.PutAsciiL(Username);
            cdefMessage.PutAsciiL(Password);
        }
    }
}
